lab-teddy/lib/storage.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const storage = {};
+
+module.exports = exports = {};
+
+exports.createCar = function(schema, car) {
+  debug('#createCar');
+  if(!schema) return Promise.reject(new Error('schema required'));
+  if(!car) return Promise.reject(new Error('car required'));
+  if(!storage[schema]) storage[schema] = {};
+
+  storage[schema][car.id] = car;
+
+  return Promise.resolve(car);
+};
+
+exports.fetchCar = function(schema, id){
+  debug('#fetchCar');
+
+  return new Promise((resolve, reject) => {
+    if(!schema) return reject(new Error('schema required'));
+    if(!id) return reject(new Error('id require'));
+
+    let schemaName = storage[schema];
+    if(!schemaName) return reject(new Error('schema not found'));
+
+    let car = schemaName[id];
+    if(!car) return reject(new Error('car not found'));
+
+    resolve(car);
+  });
+};
+// 
```

Comment:
So you'll want to make sure that you do not push up commented code. In terms of this method. Generally with puts the client would pass in specific properties that they want to change the value of. In your case, name and type. So think about how you could pass those values in and reassign with in this function.